app/services/provider_interface/hesa_data_export.rb

```diff
@@ -53,8 +51,20 @@ def export_data
 
   private
 
+    def first_degree_start(application)
+      return NO_DEGREE unless application.degrees?
+
+      year_to_iso8601 first_degree_year(application, :start_year)
+    end
+
+    def first_degree_end(application)
+      return NO_DEGREE unless application.degrees?
+
+      year_to_iso8601 first_degree_year(application, :award_year)
+    end
```

Comment:
Have we considered using a Null Object pattern for first degree? I'm not sure how it would work but it might be nicer to keep all this logic separately.

Reply:
I like this conversation ❤️ . I did consider to use the Null Object, but then I'd have to rewritten/refactor the whole class. I will invest next week for a hour and see if I can implement.

Reply:
I did a timebox and I didn't implement the nullObject per say but I moved to the degree decorator and handle there the nil cases in one place. I think this respects the same organisation that is in the code. 
Let me know your thoughts!